heim-host/src/sys/windows/platform.rs

```diff
@@ -141,22 +145,39 @@ fn rtl_get_version() -> winnt::OSVERSIONINFOEXW {
     }
 }
 
-fn get_computer_name() -> Result<String> {
-    let mut buffer: Vec<winnt::WCHAR> = Vec::with_capacity((MAX_COMPUTERNAME_LENGTH + 1) as usize);
-    let mut size: minwindef::DWORD = MAX_COMPUTERNAME_LENGTH + 1;
+fn get_value_from_get_computer_name_ex_w(kind: COMPUTER_NAME_FORMAT) -> Result<String> {
+    let mut required_size: minwindef::DWORD = 0;
+    let result = unsafe { sysinfoapi::GetComputerNameExW(kind, NULL as _, &mut required_size) };
+    if result != 0 {
+        return Err(Error::last_os_error().with_ffi("GetComputerNameEx"));
+    }
+    // required_size does not contain the trailing null byte
+    let mut size = required_size + 1;
 
-    let result = unsafe { winbase::GetComputerNameW(buffer.as_mut_ptr(), &mut size) };
+    let mut buffer: Vec<winnt::WCHAR> = vec![0; size as _]; // this ensures that buffer.len = size already (avoiding the need to resort to the unsafe `set_len` later)
+    let result = unsafe { sysinfoapi::GetComputerNameExW(kind, buffer.as_mut_ptr(), &mut size) };
     if result == 0 {
-        Err(Error::last_os_error().with_ffi("GetComputerNameW"))
-    } else {
-        unsafe {
-            buffer.set_len(size as usize + 1);
-        }
-        let str = OsString::from_wide(&buffer[..(size as usize)])
-            .to_string_lossy()
-            .to_string();
-        Ok(str)
+        return Err(Error::last_os_error().with_ffi("GetComputerNameEx"));
     }
+
+    if size > required_size {
+        // Should not happen, size "receives the number of TCHARs copied to the destination buffer, not including the terminating null character"
+        let e = std::io::Error::new(std::io::ErrorKind::Other, "Invalid value returned by GetComputerNameExW");
+        return Err(e.into());
+    }
+    // buffer[..size] is valid because buffer.len > size already
+    let str = OsString::from_wide(&buffer[..(size as usize)])
+        .to_string_lossy()
+        .to_string();
```

Comment:
Considering that `WCHAR` is basically `u16`, it looks like [`String::from_utf16`](https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf16) or `String::from_utf16_lossy` can be used here to skip the `to_string_lossy().to_string()` part.

I'm not exactly sure how often it is to have hostnames on Windows, which can't fit into UTF8, but I guess we can use `String::from_utf16_lossy` here.

Reply:
You are right, I have just pushed a commit to make this part easier